### PR TITLE
Adds functionality to the medical cyborg gripper

### PR DESCRIPTION
**Pull request by warior4356**
## What Does This PR Do
Currently, all you can do with the medical cyborg gripper is shake patients who are lying down to stand up. This PR lets you pick up, hold, and do surgery with implants, limbs, and organs using the medical cyborg's gripper. It also lets you pick up and administer pills, patches, and auto-menders. 

## Why It's Good For The Game
Currently, there is no way for a medical borg to reattach a limb, put in an implant, or give a cybernetic replacement limb.

Additionally with new crit, a medical borg is really bad at stopping someone from crashing as they are unable to deal with the ramping levels of brain damage that the cardiac arrest causes.  Once someone needs mannitol or they will start going into cardiac again, a medical borg has little ability to save the person. Also, medical borgs are bad at treating large quantities of damage besides just shoving someone into a sleeper, this should mitigate that issue somewhat, but they are still going to be far clumsier and slower at giving someone medicine that someone with hands.

People may say this just makes them doctor++. Medical cyborgs are still unable to do any surgery that requires hands, specifically removing or placing items in a cavity. Additionally, there are lots of nonsurgical tasks that doctors can do better, like minding the cloners or cyro tubes. As I said above, they are far clumsier with medicines than a doctor. Consider the following, a mid-round doctor has 5-10 fo a dozen different meds in their backpack and pockets. The medical cyborg gets what is in their hypospray (5 drugs) and then they can walk to a vending machine (no freezers for borgs), dispense one pill. Pick it up. Walk to the patient. Give them the drug.

If the medicines are controversial it would be very easily removed, but I don't think it will "invalidate doctors" personally. This PR was made to rectify their inability to reattach limbs and replace organs. The ability to hold pills, patches, and auto-menders is secondary and I am open to remove any or all of those things.

## Changelog
:cl:
add: Added the ability for medical cyborgs to pick up limbs, organs, implants, pills, patches, and auto-menders.
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Few issues with this PR:
- Borgs are intentionally worse than regular crewmates (IE: Secborgs dont have tasers or the ability to collect stripped items, engineering borgs cant use plasteel, etc). This is a substantial buff to medical borgs which really does just make them doctor++++
- This will likely conflict with the surgery PR
- Please check you havnt missed any instances of `next_step` in the code, otherwise that element of surgery will be broke

TLDR: Not in favour

**Comment by warior4356** [comment]
@AffectedArc07
I talked to farie about is surgery refactor already. He fixes the same issue I do. He can just merge over my commits and it should be fine.

I was very careful about next_step and checked every edge case I could think of, even the rare surgery that uses an open hand.

Medborgs being able to very slowly do what doctors can do, I don't think is game-breaking, like I said in the text, a doctor can carry dozens and dozens of pills and patches, able to treat any issue with just a quick grab from their bag. A medical borg has to take a trip to a vending machine for each and every dose. 

Have you tried playing a medical borg and had to deal with a crashing patient? It's quite frustrating that you can't do anything about the brain damage besides cutting their skull open.

Still like I said, the meds are optional and while I think they would be a QoL boost for borgs, are not a required part of the PR. If that's the breaking point on the PR I will remove drugs from their gripper. The main point of the PR is to let them reattach limbs and do organ surgery.

**Comment by Blackmage198** [comment]
I disagree with the "Borgs are meant to be weaker" take, as they're actually meant to be better, **roundstart**, than regular crewmembers, but then fade out in effectiveness as upgrades roll in for the crew.
For example, the Engineering Borg starts with faster tools and regenerating materials, but then Engineering gets power tools and materials roll in from mining, or like how the Security Borg is strong and it's disabler (which is, IMO, better than tasers) can fire as many times as it's battery can hold, but when security starts to get implants and the such, it falls behind the curve.

The only exception to this is the medical borg, as it's, at most, a bandaid dispenser and, rarely, a surgery machine due to it's 0% fail chance on surgery, but even that aspect is limited.

I am more than okay with these changes, as this would actually make the mediborg semi-useful beyond treating common injuries. If there are any doubts about balance, I'd suggest removing the ability to hold Automenders, as patches and pills are more overtime than them, and the fact that mediborgs wouldn't be able to store patches and pills would more than balance the ability to hold 'em.

**Comment by procdrone** [comment]
I think the only buff the borg needs is ability to access the medical fridge and ability to apply pills and patches from it. It wouldn't remove need for humans to do advanced surgery and other complicated procedures. 

And I personally think its okay. You can always assist human surgeon during such advanced surgeries or just fix everything else except stuff that needs hands. Dont think borgs are supposed to replace humans entirely. They are there to be servants not replacements

Also I do agree that its pretty dumb that mediborgs dont get automenders given its such a common tool among the normal doctors, it being makeable in autolathe and all them medical vendors. and having to recharge in cyborg to replenish stock would be pretty balanced thing, IMO

Maybe ERT borgs could be buffed more to not rely on the crew?

**Comment by Kyep** [comment]
This PR has objections from 2 maints and no votes in favor.
Most likely, it will be closed at some point.

**Comment by warior4356** [comment]
Could the objections be clarified @Kyep ? I would be happy to modify or remake the PR to be reduced in the scope of changes if there is a compromise that could be reached. It is frustrating to play as a medical cyborg and feel helpless to solve serious injuries and new crit in my opinion and I would like to at least partially address that issue.

**Comment by variableundefined** [comment]
I object on the same ground as Affected. Three objections total, closing.
